opc-ua-sdk/sdk-client/src/main/java/org/eclipse/milo/opcua/sdk/client/subscriptions/OpcUaSubscription.java

```diff
@@ -301,6 +306,123 @@ public OpcUaSubscription(
         return future.thenApply(r -> Arrays.asList(r.getRemoveResults()));
     }
 
+
+    /**
+     * Transfer the monitored items from {@code subscription} to this {@link UaSubscription}.
+     * <p>
+     * This subscription shall already have been transferred to this client's session via the Transfer Subscription
+     * service.
+     *
+     * @param fromSubscription the {@link UaSubscription} to transfer items from.
+     * @return a {@link CompletableFuture} that completes when the transfer is done.
+     */
+    CompletableFuture<Unit> transferMonitoredItems(
+        UaSubscription fromSubscription,
+        ItemTransferredCallback callback
+    ) {
+
+        List<MonitoredItemTransferRequest> monitoredItemTransfers = fromSubscription.getMonitoredItems()
+            .stream()
+            .map(
+                item ->
+                    new MonitoredItemTransferRequest(
+                        item.getReadValueId(),
+                        item.getMonitoringMode(),
+                        new MonitoringParameters(
+                            item.getClientHandle(),
+                            item.getRevisedSamplingInterval(),
+                            item.getMonitoringFilter(),
+                            item.getRevisedQueueSize(),
+                            item.getDiscardOldest()
+                        ),
+                        item.getClientHandle(),
+                        item.getMonitoredItemId()
+                    )
+            )
+            .collect(toList());
+
+        TransferMonitoredItemsRequest transferRequest = new TransferMonitoredItemsRequest(
+            monitoredItemTransfers,
+            callback
+        );
+
+        return transferMonitoredItems(newArrayList(transferRequest)).thenApply(r -> Unit.VALUE);
+    }
+
+    /**
+     * Reconstitutes the monitored items of a transferred subscription.
+     * <p>
+     * The Modify Monitored Items service will be called to synchronize the state of each monitored item.
+     * <p>
+     * This subscription shall already have been transferred to this client's session via the Transfer Subscription
+     * service.
+     *
+     * @param transferRequests a list of {@link TransferMonitoredItemsRequest}s.
+     * @return a {@link TransferMonitoredItemsResponse} that includes a list of the {@link UaMonitoredItem}s
+     * and a list of the {@link StatusCode}s for each modify result.
+     */
+    CompletableFuture<TransferMonitoredItemsResponse> transferMonitoredItems(
+        List<TransferMonitoredItemsRequest> transferRequests
```

Comment:
`transferRequests` could/should just be a single `TransferMonitoredItemsRequest`, depending on what we do on `OpcUaSubscriptionManager#transferSubscription`. See the review comment for more info.

Reply:
I believe this is the same situation as the above conversation where the intention is to allow different `ItemTransferredCallback`s for different groups of `MonitoredItemTransferRequest`s

Reply:
Is this a scenario you actually need or is it theoretical? It seems to add considerable overhead to the API signature and complexity.

Reply:
Also consider that maybe this scenario could be accomplished with multiple transfer calls instead.

Reply:
Looking at my code it appears that initially I did, but then refactored the code to utilize a single callback that incorporates a pattern match to handle the different scenarios.